packages/blockchain-wallet-v4-frontend/src/scenes/DebitCard/index.tsx

```diff
@@ -1,9 +1,20 @@
 import React from 'react'
+import { connect, ConnectedProps } from 'react-redux'
+import { withRouter } from 'react-router-dom'
+import { bindActionCreators, Dispatch } from 'redux'
+
+import { actions } from 'data'
 
 import DebitCard from './template'
 
-const DebitCardContainer = () => {
-  return <DebitCard />
-}
+const DebitCardContainer = (props: Props) => <DebitCard {...props} />
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  modalActions: bindActionCreators(actions.modals, dispatch)
+})
+
+const connector = connect(undefined, mapDispatchToProps)
+
+export type Props = ConnectedProps<typeof connector>
 
-export default DebitCardContainer
+export default withRouter(connector(DebitCardContainer))
```

Comment:
Do you need this? Can't you use a `hook`?

Reply:
Could you give an example of what you mean? I just kept the structure of another component of the wallet

Reply:
What do you use from the `withRouter`?

Reply:
you are correct, I dont need withRouter. Removed and commited